### ISSUE TITLE
set crs for raster in chapter 2 not working

### ISSUE THREAD

**Issue by wirotela** [Robinlovelace/geocompr#618]
On chapter 2 setup crs for new_raster3, it change the crs but not resolution and coordinate

2.4.3
projection(new_raster3) = "+proj=utm +zone=12 +ellps=GRS80 +towgs84=0,0,0,0,0,0,0 
                            +units=m +no_defs" # set CRS

maybe use the projectRaster()

thanks for develop the very good document.
wirote

**Comment by Robinlovelace** [comment]
Thanks for reporting this issue @wirotela. What would you expect the resolution to be? There is a section on reprojecting raster data here: https://geocompr.robinlovelace.net/reproj-geo-data.html#reprojecting-raster-geometries

I'm not sure this is a bug but if you can provide further info that may help improve the book. Thanks and great to hear you're enjoying it!

**Comment by jannes-m** [comment]
I think @wirotela is right, to reproject a raster, one should use `projectRaster()`, the `projection()` command is only setting a crs without actually transforming anything.

**Comment by Nowosad** [comment]
`"The same function, projection(), is used to set a CRS for raster objects. "` (Chapter 2) - the `projection()` function is only to set a CRS.

**Comment by jannes-m** [comment]
@Nowosad you are right, but I think the confusing part is Figure 2.14. The code somehow suggests that you could make Figure 2.14 with it but this is not the case (as you of course know), see below:

``` r
library("raster")
#> Loading required package: sp
raster_filepath = system.file("raster/srtm.tif", package = "spDataLarge")
new_raster = raster(raster_filepath)
projection(new_raster) 
#> [1] "+proj=longlat +datum=WGS84 +no_defs"
new_raster3 = new_raster
projection(new_raster3) = "+proj=utm +zone=12 +ellps=GRS80 +towgs84=0,0,0,0,0,0,0 
                          +units=m +no_defs" # set CRS
#> Warning in showSRID(uprojargs, format = "PROJ", multiline = "NO", prefer_proj
#> = prefer_proj): Discarded datum Unknown based on GRS80 ellipsoid in Proj4
#> definition
plot(new_raster)
```

![](https://i.imgur.com/EH6uYt1.png)

``` r
# this is wrong and does not correspond to Figure 2.14
plot(new_raster3)
```

![](https://i.imgur.com/8P3iVbZ.png)

``` r
# instead you would have to do
new_raster3 = new_raster
new_raster3 = projectRaster(new_raster3, crs = "+proj=utm +zone=12 +ellps=GRS80 +towgs84=0,0,0,0,0,0,0 
                          +units=m +no_defs")
#> Warning in showSRID(uprojargs, format = "PROJ", multiline = "NO", prefer_proj
#> = prefer_proj): Discarded datum Unknown based on GRS80 ellipsoid in Proj4
#> definition
plot(new_raster3)
```

![](https://i.imgur.com/XOJJhS5.png)

<sup>Created on 2021-07-27 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>

**Comment by jannes-m** [comment]
So at least we should state in the caption that we have used projectRaster(), and in the text we should additionally clarify that we have changed the projection crs string but that this resulted in a broken raster object and that one usually would use the projection() command when the crs slot of raster object is NA. Could happen with some raster formats when reading them in.

**Comment by Nowosad** [comment]
Hi @jannes-m - you are right. Sorry, I misunderstand the issue. I will add a clarifying sentence or two...

**Comment by Nowosad** [comment]
Ok, I moved the two images in the text (I think there was a longer piece of text about them that we removed in some revision), referenced them, and added a few new sentences - https://github.com/Robinlovelace/geocompr/pull/619/files. Please take a look at it @jannes-m @Robinlovelace, improve and merge it.

**Comment by Robinlovelace** [comment]
Fixed in #619
